File: Practice/AllDomains/CoreCS/DataStructures/Arrays/DynamicArray.cs
/*	author:@shivkrthakur  */
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
class Solution {
    static void Main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution */
        int [] NQ = Console.ReadLine().Split().Select(Int32.Parse).ToArray();
        int n = NQ[0], Q = NQ[1];
        int [][] seqList = new int[n][];

        int lastAns = 0;
        for(int i = 0; i < Q; i++)
        {
            int [] qxy = Console.ReadLine().Split().Select(Int32.Parse).ToArray();
            int q = qxy[0], x = qxy[1], y = qxy[2];
            int index;
            int [] seq;
            switch(q)
            {
                case 1:
                    index = (x ^ lastAns) % n;
                    seq = seqList[index];
                    if(seq == null ) { seq = new int[] { y }; seqList[index] = seq; }
                    else
                    {
                        Array.Resize(ref seq, seq.Length + 1);
                        seq[seq.Length - 1] = y;
                        seqList[index] = seq;
                    }
                    break;

                case 2:
                    index = (x ^ lastAns) % n;
                    seq = seqList[index];
                    lastAns = y % seq.Length;
                    lastAns = seq[lastAns];
                    Console.WriteLine($"{lastAns}");
                    break;
            }
        }
    }
}

File: Practice/AllDomains/CoreCS/DataStructures/Arrays/DynamicArray.js
/*	author:@shivkrthakur  */
var index = 0;

function readLine(input){
    return input[index++];
}

function processData(input) {
    //Enter your code here
    input = input.split('\n');
    var ip = readLine(input).split(' ').map(Number);
    var N = ip[0], Q = ip[1];
    var op = [];
    for(var k = 0; k < N; k++) op[k] = new Array();
    var lastAnswer = 0;
    for(var i = 0; i < Q; i++){
        var arr = readLine(input).split(' ').map(Number);
        var x = arr[1];
        var y = arr[2];
        x = (x ^ lastAnswer) % N;
        if(arr[0] == 1){
            op[x].push(y);
        }
        else{
            var seq = op[x];
            lastAnswer = seq[y % seq.length];
            console.log(lastAnswer);
        }
    }
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});
